frappe/public/js/frappe/form/multi_select_dialog.js

```diff
@@ -128,7 +98,80 @@ frappe.ui.form.MultiSelectDialog = Class.extend({
 		this.dialog.show();
 	},
 
-	bind_events: function() {
+	make_dialog_header: function () {
+		let fields = [];
+		let me = this;
+		this.col_0 = [
+			{
+				fieldtype: "Data",
+				label: __("Search Term"),
+				fieldname: "search_term"
+			}
+		];
+		this.col_1 = [];
+		this.col_2 = [];
+
+		// setters can be defined as a dict or a list of fields
+		// setters define the additional filters that get applied
+		// for selection
+
+		// CASE 1: DocType name and fieldname is the same, example "customer" and "customer"
+		// setters define the filters applied in the modal
+		// if the fieldnames and doctypes are consistently named,
+		// pass a dict with the setter key and value, for example
+		// {customer: [customer_name]}
+
+		// CASE 2: if the fieldname of the target is different,
+		// then pass a list of fields with appropriate
+
+		if ($.isArray(this.setters)) {  // need to figure out
+			for (let df of this.setters) {
+				fields.push(df, { fieldtype: "Column Break" });
+			}
+		} else {
+			Object.keys(this.setters).forEach(function (setter, index) {
+				// x['group_' + cstr((index+1)%4)].push(setter);
+				let df_prop = frappe.meta.docfield_map[me.doctype][setter];
+				// this[]push({fieldtype: "Column Break"});
+				me['col_' + cstr((index + 1) % 3)].push({
```

Comment:
Don't attach it to a `this` maintain a local variable

Reply:
done